evennia/contrib/mapbuilder.py

```diff
@@ -0,0 +1,458 @@
+For example:
+
+    # mygame/commands/default_cmdsets.py
+
+    from evennia.contrib import mapbuilder
+
+    ...
+
+    self.add(mapbuilder.CmdMapBuilder())
+
+You then call the command in-game using the path to the MAP and MAP_LEGEND vars
+The path you provide is relative to the evennia or mygame folder.
+
+Usage:
+    @mapbuilder[/switch] <path.to.file.MAPNAME> <path.to.file.MAP_LEGEND>
+
+Switches:
+  one - execute build instructions once without automatic exit creation.
+  two - execute build instructions twice without automatic exit creation.
+
+Example:
+    @mapbuilder world.gamemap.MAP world.maplegend.MAP_LEGEND
+    @mapbuilder evennia.contrib.mapbuilder.EXAMPLE2_MAP EXAMPLE2_LEGEND
+            (Legend path defaults to map path)
+
+Below are two examples showcasing the use of automatic exit generation and
+custom exit generation. Whilst located, and can be used, from this module for
+convenience The below example code should be in mymap.py in mygame/world.
+
+"""
+
+from django.conf import settings
+from evennia.utils import utils
+
+# ---------- EXAMPLE 1 ---------- #
+# @mapbuilder evennia.contrib.mapbuilder.EXAMPLE1_MAP EXAMPLE1_LEGEND
+
+# -*- coding: utf-8 -*-
+
+# Add the necessary imports for your instructions here.
+from evennia import create_object
+from typeclasses import rooms, exits
+from evennia.utils import utils
+from random import randint
+import random
+
+# A map with a temple (▲) amongst mountains (n,∩) in a forest (♣,♠) on an
+# island surrounded by water (≈). By giving no instructions for the water
+# characters we effectively skip it and create no rooms for those squares.
+EXAMPLE1_MAP = """\
+≈≈≈≈≈
+≈♣n♣≈
+≈∩▲∩≈
+≈♠n♠≈
+≈≈≈≈≈
+"""
+
+
+def build_forest(x, y, **kwargs):
+    """A basic example of build instructions. Make sure to include **kwargs
+    in the arguments and return an instance of the room for exit generation."""
+
+    # Create a room and provide a basic description.
+    room = create_object(rooms.Room, key="forest" + str(x) + str(y))
+    room.db.desc = "Basic forest room."
+
+    # Send a message to the player
+    kwargs["caller"].msg("Forest Room Created.")
+
+    # This is generally mandatory.
+    return room
+
+
+def build_mountains(x, y, **kwargs):
+    """A room that is a little more advanced"""
+
+    # Create the room.
+    room = create_object(rooms.Room, key="mountains" + str(x) + str(y))
+    
+    # Generate a description by randomly selecting an entry from a list.
+    room_desc = ["Mountains as far as the eye can see",
+                 "Your path is surrounded by sheer cliffs",
+                 "Haven't you seen that rock before?"]
+    room.db.desc = random.choice(room_desc)
+    
+    # Create a random number of objects to populate the room.
+    for i in xrange(randint(0, 3)):
+        rock = create_object(key="Rock", location=room)
+        rock.db.desc = "An ordinary rock."
+
+    # Send a message to the player
+    kwargs["caller"].msg("Mountain Room Created.")
```

Comment:
One thing I noticed with this was how to get to the area just created. I know this is just an example and that people can change the input as they want, but since all these rooms are named the same, it might set a good precedent to include the #dbref of the newly created room in this return text, so one can easily jump to it and check things out afterwards.